From 3d3364efdcd4bf765e66d193fae2573981d4cd79 Mon Sep 17 00:00:00 2001
From: Tom Hughes <tomhughes@google.com>
Date: Fri, 7 Aug 2020 15:35:57 -0700
Subject: [PATCH 14/15] Fix compilation error when building with C++17

noexcept needs to be on definition in addition to the declaration

Bug: 162949739
Test: Builds
Change-Id: I3dbd316c81cdb5f5e7e2531e730cdd71f8ed29cb
---
 android_keymaster/keymaster_stl.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/android_keymaster/keymaster_stl.cpp b/android_keymaster/keymaster_stl.cpp
index 62c486c..5ca9ed5 100644
--- a/android_keymaster/keymaster_stl.cpp
+++ b/android_keymaster/keymaster_stl.cpp
@@ -24,10 +24,10 @@ struct nothrow_t {};
 
 extern const std::nothrow_t __attribute__((weak)) std::nothrow = {};
 
-void* __attribute__((weak)) operator new(size_t __sz, const std::nothrow_t&) {
+void* __attribute__((weak)) operator new(size_t __sz, const std::nothrow_t&) _NOEXCEPT {
     return malloc(__sz);
 }
-void* __attribute__((weak)) operator new[](size_t __sz, const std::nothrow_t&) {
+void* __attribute__((weak)) operator new[](size_t __sz, const std::nothrow_t&) _NOEXCEPT {
     return malloc(__sz);
 }
 
-- 
2.17.1

